### ISSUE TITLE
RasterOperation with GeoTiff fails when operationType='image'

### ISSUE THREAD

**Issue by valluzzi** [openlayers/openlayers#14406]
**Describe the bug**
Trying to reproduce something like region growing example but using GeoTiff
https://openlayers.org/en/latest/examples/region-growing.html
I have an error like  **tile.getImage is not a function**
Raster operation works well if operationType is "pixel" but when I switch to operationType:"image" it does not work.
I dont know if it's a bug or my error.    

**To Reproduce**
```

import {Raster as RasterSource, GeoTIFF} from 'ol/source'
import Map from 'ol/Map';
import { WebGLTile as TileLayer } from 'ol/layer'
import View from 'ol/View';
import OSM from 'ol/source/OSM'
import proj4 from 'proj4';
import {register} from 'ol/proj/proj4';
import ImageLayer from 'ol/layer/Image';
import filetif from './COPERNICUS_EUDEM_165252.tif'

proj4.defs(
  'EPSG:32633','+proj=utm +zone=33 +ellps=WGS84 +datum=WGS84 +units=m +no_defs'
);
register(proj4);

const legend = [
  -9999,'#00000000',  //Nodata=-9999 transparent
  0, "#000000",    //MinValue   Black
  150,"#ffffff"   //MaxValue    White
];

const BlackToWhite = {

  color: [
    'case',
    ['==', ['band', 2], 0],  //se la banda2 == 0 (indica i nodata)
    '#00000000',             //allora pixel trasparente
    ['interpolate', ['linear'], ['band', 1], ...legend ],  //altrimenti interploazione lineare banda1 colori della legenda
],
}

const elev = new GeoTIFF({
  sources: [
    {
      url: filetif,
      projection: 'EPSG:32633',
    },
  ],
  normalize: false,   //important
  interpolate: false, //important
});

const osm = new TileLayer({
  visible: true,
  preload: Infinity,
  source: new OSM()
})

const raster = new RasterSource({
  sources: [elev],
  //operation: (pixels, data)=>{return [Math.round(Math.random()*255),0,0,255]}, //this works with operationType:"pixel"
  operationType: 'image',
  operation: (inputs, data) => {   //silly operation that do nothing 
     const image = inputs[0];
     let n = image.width
     let m = image.height
     return {data: image.data, width: n, height: m}}
  });

const dem = new ImageLayer({
  source: raster,
  style: BlackToWhite
});

const map = new Map({
  target: 'map',
  layers: [

      osm,

      dem

],
  view: new View({

      projection: 'EPSG:32633',
      center:[306821.2,4879578.4],
      zoom:13

  })
});
```

[COPERNICUS_EUDEM_165252.zip](https://github.com/openlayers/openlayers/files/10285704/COPERNICUS_EUDEM_165252.zip)

**Comment by mike-000** [comment]
You need to pass a GeoTIFF source to the RasterSource as a WebGLTileLayer otherwise the RasterSource will use the supplied source with a canvas TileLayer.  Also your style should be added to that WebGLTileLayer, not the ImageLayer.

```
const elev = new TileLayer({
  source: new GeoTIFF({
    sources: [
      {
        url: filetif,
      },
    ],
    normalize: false,   //important
    interpolate: false, //important
  },
  style: BlackToWhite
});

const raster = new RasterSource({
  sources: [elev],
  //operation: (pixels, data)=>{return [Math.round(Math.random()*255),0,0,255]}, //this works with operationType:"pixel"
  operationType: 'image',
  operation: (inputs, data) => {   //silly operation that do nothing 
     const image = inputs[0];
     let n = image.width
     let m = image.height
     return {data: image.data, width: n, height: m}}
  });

const dem = new ImageLayer({
  source: raster,
});
```

Similar to https://deploy-preview-13365--ol-site.netlify.app/examples/webgl-interpolation.html (although the WebGLTileLayer there does not use a GeoTIFF source)

**Comment by valluzzi** [comment]
Thank a lot, It' works!

**Comment by gismatthew** [comment]
@valluzzi, @mike-000  I'm doing similar work and this thread helps a lot. I still notice abnormal behaviours regarding RasterOperation on GeoTiff source, with your data and code ( operationType as both `image` and `pixel` ):
- endless loop of raster operation
- if I show both `elev` layer (TileLayer) and `dem` layer (ImageLayer), `elev` layer is not visible (flickers when being dragged then disappear when the cursor is released) and `dem` layer is invisible as as well because:
- within the operation function, pixel values are always `0`s, no matter whether it's `image` or `pixel` type operation
- I checked the library's source code (`Raster.js`) and noticed that the pixel values are actually the rendered `RGBA` values (e.g. rendered colour of 'elev') instead of the original/raw value from the GeoTiff. This is not what I originally thought and undocumented, nonetheless I can bypass this limitation working with rendered values, only if I can make it work.

I wonder how the current code works for you and any ideas are appreciated!

**Comment by valluzzi** [comment]
Yes, I think my collegue make It works. I Will answer you as soon as possible

**Comment by nmusso** [comment]
@gismatthew Hi, try to have a look at #14408 and #14409, I had similar errors and seems like an underlying problem of OpenLayers

**Comment by mike-000** [comment]
If you have opacity in the rendered GeoTIFF you will need to use dev version of OpenLayers, the fix for #14408 and #14409 has been added since the 7.2.2 release.
